### ISSUE TITLE
3.12 support

### ISSUE THREAD

**Issue by samuelcolvin** [PyO3/maturin#1627]
See https://github.com/pydantic/pydantic-core/pull/629 It seems like trying to build for python 3.12 fails.

Miraculously our tests are passing, hence it would be good to build binaries so we can test the rest of pydantic.

<details>
<summary>Error output</summary>

```
📦 Including license file "/home/runner/work/pydantic-core/pydantic-core/LICENSE"
🍹 Building a mixed python/rust project
🔗 Found pyo3 bindings
💥 maturin failed
  Caused by: Failed to find a CPython 3.12 interpreter
Error: The process '/usr/bin/docker' failed with exit code 1
Error: The process '/usr/bin/docker' failed with exit code 1
    at ExecState._setResult (/home/runner/work/_actions/PyO3/maturin-action/v1/dist/index.js:1702:25)
    at ExecState.CheckComplete (/home/runner/work/_actions/PyO3/maturin-action/v1/dist/index.js:1685:18)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/PyO3/maturin-action/v1/dist/index.js:1579:27)
    at ChildProcess.emit (node:events:527:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
```

</details>

**Comment by samuelcolvin** [comment]
on reflection, I guess this should be moved to https://github.com/PyO3/maturin-action?

**Comment by messense** [comment]
It's fine to leave it here, this requires quite some coordinations in different repositories.

**Comment by messense** [comment]
Should be ready in next release.

**Comment by messense** [comment]
Released in v1.0.1.
